velox/functions/sparksql/StringToMap.h

```diff
@@ -32,71 +32,103 @@ struct StringToMapFunction {
       const arg_type<Varchar>& input,
       const arg_type<Varchar>& entryDelimiter,
       const arg_type<Varchar>& keyValueDelimiter) {
-    VELOX_USER_CHECK_EQ(
-        entryDelimiter.size(), 1, "entryDelimiter's size should be 1.");
-    VELOX_USER_CHECK_EQ(
-        keyValueDelimiter.size(), 1, "keyValueDelimiter's size should be 1.");
-
-    callImpl(
-        out,
-        toStringView(input),
-        toStringView(entryDelimiter),
-        toStringView(keyValueDelimiter));
+    VELOX_USER_CHECK_GE(
+        entryDelimiter.size(), 1, "The entry delimiter should not be empty.");
+    VELOX_USER_CHECK_GE(
+        keyValueDelimiter.size(),
+        1,
+        "The keyValue delimiter should not be empty.");
+    callImpl(out, input, entryDelimiter, keyValueDelimiter);
   }
 
  private:
-  static std::string_view toStringView(const arg_type<Varchar>& input) {
-    return std::string_view(input.data(), input.size());
+  /// @brief Retrieve the index of the pattern string
+  ///
+  /// This function retrieves all subscripts in the target string that match the
+  /// start and end of the pattern
+  ///
+  /// @param targetStr The target string to search for
+  /// @param pattern Search pattern.
+  std::vector<std::pair<int, int>> findPatternIndex(
+      const StringView& targetStr,
+      const StringView& pattern) {
+    std::vector<std::pair<int, int>> result;
+    auto patternStr = pattern.str();
+    patternStr.insert(patternStr.begin(), '(');
+    patternStr.push_back(')');
```

Comment:
Perhaps follow below implementation to avoid creating intermediate strings, and use cache if hitting the same regex pattern.

https://github.com/facebookincubator/velox/pull/10248/files#diff-4bcf5e9c7cba9c7caeeb19e82ea2f1170f73b81f5172f9808de419515b9f26c8R130-R131